packages/core/core/src/public/BundleGraph.js

```diff
@@ -210,3 +221,66 @@ function findBundlesWithAsset(
       return node.value;
     });
 }
+
+function bundleNodeToBundleReferenceNode(
+  bundleNode: BundleNode
+): BundleReferenceNode {
+  return {
+    id: bundleNode.id,
+    type: 'bundle_reference',
+    value: new BundleReference(bundleNode.value)
+  };
+}
+
+function mergeBundleGraphIntoBundleAssetGraph(
+  bundleAssetGraph: Graph<AssetGraphNode>,
+  bundleGraph: Graph<BundleGraphNode>
+): void {
+  for (let [, node] of bundleGraph.nodes) {
+    if (node.type === 'bundle') {
+      bundleAssetGraph.addNode(bundleNodeToBundleReferenceNode(node));
+    } else {
+      bundleAssetGraph.addNode(node);
+    }
+  }
+
+  for (let edge of bundleGraph.edges) {
+    bundleAssetGraph.addEdge(edge);
+  }
+}
+
+class BundleReference {
+  #bundle;
```

Comment:
We need to wrap instead of copy as bundles can be mutated after they're created (bundler, namers, etc.)